homeassistant/components/overkiz/config_flow.py

```diff
@@ -59,15 +57,30 @@ async def async_validate_input(self, user_input):
 
         async with TahomaClient(username, password, api_url=endpoint) as client:
             await client.login()
-            return self.async_create_entry(title=username, data=user_input)
+
+            if self._reauth_entry is None:
+                return self.async_create_entry(title=username, data=user_input)
+
+            # Modify existing entry in reauth scenario
+            self.hass.config_entries.async_update_entry(
+                self._reauth_entry, data=user_input
+            )
+
+            await self.hass.config_entries.async_reload(self._reauth_entry.entry_id)
+
+            return self.async_abort(reason="reauth_successful")
 
     async def async_step_user(self, user_input=None):
         """Handle the initial step via config flow."""
         errors = {}
 
         if user_input:
             await self.async_set_unique_id(user_input.get(CONF_USERNAME))
-            self._abort_if_unique_id_configured()
+
+            if not self._reauth_entry:
+                self._abort_if_unique_id_configured()
+            elif self._reauth_entry.unique_id != self.unique_id:
+                return self.async_abort(reason="already_configured")
```

Comment:
Why is it already configured if the unique ID is different ?

Reply:
If the username is different, the user is not re-authenticating and so we should create a new config entry. Or, in that case maybe we should not allow the user to enter a username to begin with. Just show a password field. You can show the chosen hub / username as labels instead.

Reply:
The user case from `smarttub` was an api that allowed you to change your email address and retain the unique (account) id.

Does the service allow changing the username and retaining the same unique id?

Reply:
@bdraco unfortunately, currently the unique id in the Config Flow is set to the email address. Thinking of this, it would be a good thing to retrieve the gateways in the Config Flow and set this as their unique identifier. This is the same identifier as we use for DHCP discovery.

However, we can only pull this after a successful login, thus I am not sure if this will work in all scenario's.

> If the username is different, the user is not re-authenticating and so we should create a new config entry. Or, in that case maybe we should not allow the user to enter a username to begin with. Just show a password field. You can show the chosen hub / username as labels instead.

That would indeed be a good one to implement. I haven't seen reauth examples yet where they add the information as labels and only allow changing the password.

Reply:
As long as you get the same unique id from discovery and after successful login it should work for most/all use cases as that's a common pattern we have in many integrations

Reply:
Fixed in bde8060. We now use the gateway id in the config flow, which is the best unique identifier, instead of the users email address). Thanks for the suggestion.